scripts/irods/lib.py

```diff
@@ -684,14 +688,16 @@ def replica_exists(session, logical_path, replica_number):
     return 'CAT_NO_ROWS_FOUND' not in out
 
 
-def replica_exists_on_resource(session, logical_path, resource_name):
-    out = session.run_icommand(['iquest',
-        "select DATA_ID where COLL_NAME = '{}' and DATA_NAME = '{}' and DATA_RESC_NAME = '{}'"
-        .format(os.path.dirname(logical_path),
-                os.path.basename(logical_path),
-                resource_name)])[0]
-
-    return 'CAT_NO_ROWS_FOUND' not in out
+def replica_exists_on_resource(session, logical_path, resource_name, zone_name=None):
+    query = "select DATA_ID where COLL_NAME = '{}' and DATA_NAME = '{}' and DATA_RESC_NAME = '{}'".format(
+        os.path.dirname(logical_path),
+        os.path.basename(logical_path),
+        resource_name)
+    if zone_name:
+        command = ['iquest', '-z', zone_name, '%s', query]
+    else:
+        command = ['iquest', '%s', query]
+    return 'CAT_NO_ROWS_FOUND' not in session.run_icommand(command)[0]
```

Comment:
I found this line harder to read than I expected.   Perhaps a comment for the next reader?

Reply:
Would it help to split up the line? Like this...
```suggestion
    out,_,_ = session.run_icommand(command)
    return 'CAT_NO_ROWS_FOUND' not in out
```

Reply:
i think it's the negative that got me... so either split them or a comment about what will happen for the True/False cases.

or... you know, assume the reader can read :P

Reply:
"that there no results" -> "that no results"

otherwise, this is very helpful - yes.  please.